src/compiler/checker.ts

```diff
@@ -18172,7 +18176,7 @@ namespace ts {
         }
 
         // Return contextual type of parameter or undefined if no contextual type is available
-        function getContextuallyTypedParameterType(parameter: ParameterDeclaration): Type | undefined {
+        function getContextuallyTypedParameterType(parameter: ParameterDeclaration, forCache?: boolean): Type | undefined {

```

Comment:
this function is only called twice so I would make `forCache` required.

monomorphism!